packages/toolpad-app/src/toolpad/AppEditor/PageTitleEditor.tsx

```diff
@@ -0,0 +1,66 @@
+import { TextField } from '@mui/material';
+import * as React from 'react';
+import { AppDom, PageNode } from '../../appDom';
+import { useDomApi } from '../AppState';
+import { update } from '../../utils/immutability';
+
+interface PageTitleEditorProps {
+  node: PageNode;
+}
+
+export default function PageTitleEditor({ node }: PageTitleEditorProps) {
+  const domApi = useDomApi();
+  const [pageTitleInput, setPageTitleInput] = React.useState(node.attributes.title);
+  const [isValid, setIsValid] = React.useState(true);
+
+  const handlePageTitleChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => setPageTitleInput(event.target.value),
+    [],
+  );
+
+  const handleCommit = React.useCallback(() => {
+    if (!isValid) {
+      return;
+    }
+    domApi.update((appDom: AppDom) => {
+      return update(appDom, {
```

Comment:
You can use [`setNamespacedProp`](https://github.com/mui/mui-toolpad/blob/eacf6f9cdbce81868c44d38055e398a6980d7022/packages/toolpad-app/src/appDom/index.ts#L713C35-L713C35) for this.

Reply:
@Kirera-Wainaina Sorry, I meant [`setNodeNamespacedProp`](https://github.com/mui/mui-toolpad/blob/eacf6f9cdbce81868c44d38055e398a6980d7022/packages/toolpad-app/src/appDom/index.ts#L739)